test_unstructured/partition/html/test_parser.py

```diff
@@ -428,24 +660,574 @@ def it_generates_text_segments_for_its_text_and_children_and_tail(
+            (
+                "<strong><q>aaa</q><p>bbb</p>ccc<s>ddd</s>eee</strong>",
+                "b",
+                [
+                    TextSegment(
+                        "aaa", {"emphasized_text_contents": "aaa", "emphasized_text_tags": "b"}
+                    ),
+                    Title("bbb"),
+                    TextSegment(
+                        "ccc", {"emphasized_text_contents": "ccc", "emphasized_text_tags": "b"}
+                    ),
+                    TextSegment(
+                        "ddd", {"emphasized_text_contents": "ddd", "emphasized_text_tags": "b"}
+                    ),
+                    TextSegment(
+                        "eee", {"emphasized_text_contents": "eee", "emphasized_text_tags": "b"}
+                    ),
+                ],
+            ),
+        ],
+    )
+    def and_it_generates_elements_for_its_block_children(
+        self, html_text: str, inside_emphasis: str, expected_value: list[TextSegment | Element]
+    ):
+        e = etree.fromstring(html_text, html_parser).xpath(".//body")[0][0]
+        assert list(e._iter_child_text_segments(inside_emphasis)) == expected_value
+
+    # -- ._iter_text_segments_from_block_tail_and_phrasing() --------------
+
+    @pytest.mark.parametrize(
+        ("html_text", "emphasis", "expected_value"),
+        [
+            # -- no tail and no contiguous phrasing produces no text-segments --
+            ("<cite><p/></cite>", "", []),
+            # -- tail produces a text-segment --
+            ("<cite><p/>aaa</cite>", "", [TextSegment("aaa", {})]),
+            # -- contiguous phrasing produces a text-segment --
+            ("<cite><p/><s>aaa</s></cite>", "", [TextSegment("aaa", {})]),
+            # -- tail of contiguous phrasing also produces a text-segment --
+            ("<bdi><p/><s>aaa</s>bbb</bdi>", "", [TextSegment("aaa", {}), TextSegment("bbb", {})]),
+            # -- nested phrasing produces a text-segment --
+            (
+                "<sub><p/>aaa<s>bbb<q>ccc</q>ddd</s>eee</sub>",
+                "",
+                [
+                    TextSegment("aaa", {}),
+                    TextSegment("bbb", {}),
+                    TextSegment("ccc", {}),
+                    TextSegment("ddd", {}),
+                    TextSegment("eee", {}),
+                ],
+            ),
+            # -- and emphasis is added to each text-segment when specified --
+            (
+                "<strong><p/>aaa<s>bbb<i>ccc</i>ddd</s>eee</strong>",
+                "b",
+                [
+                    TextSegment(
+                        "aaa", {"emphasized_text_contents": "aaa", "emphasized_text_tags": "b"}
+                    ),
+                    TextSegment(
+                        "bbb", {"emphasized_text_contents": "bbb", "emphasized_text_tags": "b"}
+                    ),
+                    TextSegment(
+                        "ccc", {"emphasized_text_contents": "ccc", "emphasized_text_tags": "bi"}
```

Comment:
@scanny - You might have told me before, but will this always be `"bi"` or would it be `"ib"` if the nesting of the tags were reversed?

Reply:
It will always be `"bi"`. Seemed like canonicalizing it was the right thing to do, just to avoid unnecessary complexity. I'm not sure we could say which came "first" when both are applied because the effect is the same from a visual perspective.

Reply:
https://github.com/Unstructured-IO/unstructured/blob/2a12131b4e452b40e1597be888dcfb46c882f9aa/unstructured/partition/html/parser.py#L770-L779

Reply:
Yeah I think having a canonical representation is the way to go, just wanted to double check. Especially since, as you mentioned, it doesn't make any difference visually.